plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/IsNullPushdownDataTypeTest.java

```diff
@@ -0,0 +1,241 @@
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public final class IsNullPushdownDataTypeTest
+{
+    private static final int SPECIAL_COLUMNS = 1;
+
+    public static IsNullPushdownDataTypeTest create(PushdownImplementation pushdownImplementation)
+    {
+        return new IsNullPushdownDataTypeTest(pushdownImplementation);
+    }
+
+    private final List<TestCase> testCases = new ArrayList<>();
+    private final PushdownImplementation pushdownImplementation;
+
+    private IsNullPushdownDataTypeTest(PushdownImplementation pushdownImplementation)
+    {
+        this.pushdownImplementation = pushdownImplementation;
+    }
+
+    public IsNullPushdownDataTypeTest addRoundTrip(String literal)
+    {
+        return addRoundTrip(literal, literal);
+    }
+
+    public IsNullPushdownDataTypeTest addRoundTrip(String inputLiteral, String expectedLiteral)
+    {
+        testCases.add(new TestCase(Optional.empty(), Optional.empty(), inputLiteral, Optional.empty(), expectedLiteral));
+        return this;
+    }
+
+    public IsNullPushdownDataTypeTest addRoundTrip(String inputType, String literal, Type expectedType)
+    {
+        return addRoundTrip(inputType, literal, expectedType, literal);
+    }
+
+    public IsNullPushdownDataTypeTest addRoundTrip(String inputType, String inputLiteral, Type expectedType, String expectedLiteral)
+    {
+        addRoundTrip(Optional.empty(), inputType, inputLiteral, expectedType, expectedLiteral);
+        return this;
+    }
+
+    public IsNullPushdownDataTypeTest addRoundTrip(String columnName, String inputType, String inputLiteral, Type expectedType, String expectedLiteral)
+    {
+        addRoundTrip(Optional.of(columnName), inputType, inputLiteral, expectedType, expectedLiteral);
+        return this;
+    }
+
+    public IsNullPushdownDataTypeTest addRoundTrip(Optional<String> columnName, String inputType, String inputLiteral, Type expectedType, String expectedLiteral)
+    {
+        testCases.add(new TestCase(columnName, Optional.of(inputType), inputLiteral, Optional.of(expectedType), expectedLiteral));
+        return this;
+    }
+
+    public IsNullPushdownDataTypeTest execute(QueryRunner queryRunner, DataSetup dataSetup)
+    {
+        return execute(queryRunner, queryRunner.getDefaultSession(), dataSetup);
+    }
+
+    public IsNullPushdownDataTypeTest execute(QueryRunner queryRunner, Session session, DataSetup dataSetup)
+    {
+        checkState(!testCases.isEmpty(), "No test cases");
+        for (int specialColumn = 0; specialColumn < SPECIAL_COLUMNS; specialColumn++) {
+            checkArgument(!"NULL".equalsIgnoreCase(testCases.get(specialColumn).inputLiteral()));
+        }
+
+        try (TemporaryRelation temporaryRelation = dataSetup.setupTemporaryRelation(unmodifiableList(testCases))) {
+            verifySelect(queryRunner, session, temporaryRelation);
```

Comment:
We do test them in `DataTypeMappingTest` right ?

Reply:
The test itself is not needed, the check was left for consistency of the DataSetup,
as the goal is not to think about preparation of the `IsNullPushdownDataTypeTest` but mostly for blind copy-paste of cases from `DataTypeMappingTest`.